aodndata/anfog/classifiers.py

```diff
@@ -0,0 +1,127 @@
+                position_txtfile = handler.file_collection.filter_by_attribute_regex('name', '.*position_summary.txt$')
+                if (png is not None) or (position_txtfile is not None):
+                    return 'RT'
+                else:
+                    raise InvalidFileContentError(
+                        "Missing ancillary files in ZIP archive {name}".format(
+                            name=os.path.basename(handler.input_file)))
+
+    @classmethod
+    def get_platform(cls, filename):
+        """ work out platform type from filename
+            DSTG and NRL deployments are slocum glider only
+            ANFOG deployments are either seaglider or slocum glider """
+        if (re.match(cls.DSTG_REGEX, filename)) or (re.match(cls.NRL_REGEX, filename)):
+            platform = 'slocum_glider'
+        else:
+            fields = cls._get_file_name_fields(filename)
+            platform_id = fields[4][:2]
+            platform = cls.PLATFORM_CODES[platform_id]
+
+        return platform
+
+    @classmethod
+    def get_deployment_code(cls, src_file):
+        """Depending on data mode :
+           DM :  get deployment code from netcdf global attributes directly
+               DSTG : no attribute deployment_code, extract deployment code from title instead
+           RT :exctract deployment code from title
+        """
+        if re.match(cls.DSTG_REGEX, src_file.name) or re.match(cls.ANFOG_RT_REGEX, src_file.name):
+            title = cls._get_nc_att(src_file.src_path, 'title')
+            deployment_code = title.split()[-1]
+            if deployment_code == 'mission':
+                raise InvalidFileContentError(
+                    "Missing deployment code in {file} ".format(file=os.path.basename(src_file.name)))
+
+        elif re.match(cls.ANFOG_DM_REGEX, src_file.name):
+            deployment_code = cls._get_nc_att(src_file.src_path, 'deployment_code')
+        else:
+            raise InvalidFileNameError(
+                "Invalidfile name {file} ".format(file=os.path.basename(src_file.name)))
+
+        return deployment_code
+
+    @classmethod
+    def set_attributes_per_file_type(cls, filecollection, destination, path):
+        """
+        Set check_type,publish_type and dest_path or archive_path attribute
+           to a collection of file of the same type/format according to its destination
+           destination is either 'S3upload' or 'archive'
+        """
+        for f in filecollection:
+            f.check_type = PipelineFileCheckType.FORMAT_CHECK
+            if destination == 'S3upload':
+                f.publish_type = PipelineFilePublishType.UPLOAD_ONLY
+                f.dest_path = os.path.join(path, f.name)
+            elif destination == 'archive':
+                f.publish_type = PipelineFilePublishType.ARCHIVE_ONLY
+                f.archive_path = os.path.join(path, f.name)
+
+    @classmethod
+    def dest_path(cls, src_file, data_mode):
```

Comment:
I think this and the "set_data_mode" method could be combined/refactored in the handler such that we could get rid of all the manual setting of dest_paths everywhere and make the dest_path function work in the intended manner.